searchlib/src/vespa/searchlib/diskindex/posting_list_cache.h

```diff
@@ -20,6 +20,44 @@ class PostingListCache : public IPostingListCache {
     std::unique_ptr<Cache> _cache;
     std::unique_ptr<BitVectorCache> _bitvector_cache;
 public:
+    class CacheSizingParams {
+        size_t _posting_max_bytes   = 0;
+        size_t _bitvector_max_bytes = 0;
+        double _posting_slru_protected_ratio   = 0.0; // [0, 1]
+        double _bitvector_slru_protected_ratio = 0.0; // [0, 1]
+    public:
+        constexpr CacheSizingParams() noexcept = default;
+        CacheSizingParams(size_t posting_max_bytes, size_t bitvector_max_bytes,
+                          double posting_slru_protected_ratio, double bitvector_slru_protected_ratio) noexcept
+            : _posting_max_bytes(posting_max_bytes),
+              _bitvector_max_bytes(bitvector_max_bytes),
+              _posting_slru_protected_ratio(std::min(std::max(posting_slru_protected_ratio, 0.0), 1.0)),
+              _bitvector_slru_protected_ratio(std::min(std::max(bitvector_slru_protected_ratio, 0.0), 1.0))
+        {
+        }
+
+        [[nodiscard]] size_t posting_slru_protected_bytes() const noexcept {
+            if (_posting_slru_protected_ratio <= 0) {
+                return 0;
+            }
+            return static_cast<size_t>(static_cast<double>(_posting_max_bytes) * _posting_slru_protected_ratio);
+        }
+        [[nodiscard]] size_t posting_slru_probationary_bytes() const noexcept {
+            return _posting_max_bytes - posting_slru_protected_bytes();
+        }
+
+        [[nodiscard]] size_t bitvector_slru_protected_bytes() const noexcept {
+            if (_bitvector_slru_protected_ratio <= 0) {
+                return 0;
+            }
+            return static_cast<size_t>(static_cast<double>(_bitvector_max_bytes) * _bitvector_slru_protected_ratio);
+        }
+        [[nodiscard]] size_t bitvector_slru_probationary_bytes() const noexcept {
+            return _bitvector_max_bytes - posting_slru_protected_bytes();
```

Comment:
Should this use `bitvector_slru_protected_bytes()` instead?

Reply:
Definitely. Fixed.